php-checks/src/main/java/org/sonar/php/checks/MultilineBlocksCurlyBracesCheck.java

```diff
@@ -91,39 +94,52 @@ private static StatementTree getLastStatement(IfStatementTree ifStatement) {
     return statement;
   }
 
-  private void checkStatement(StatementTree firstInnerStatement, int nestingStatementNum, List<StatementTree> statements) {
+  private void checkStatement(Tree headerStart, Tree headerEnd, StatementTree firstInnerStatement, int nestingStatementNum, List<StatementTree> statements) {
     if (firstInnerStatement.is(Kind.BLOCK)) {
       return;
     }
+    int headerLine = ((PHPTree) headerEnd).getLastToken().line();
+    int headerColumn = ((PHPTree) headerStart).getFirstToken().column();
+    List<StatementTree> badIndentationStatements = badIndentationStatements(headerLine, headerColumn, firstInnerStatement, nestingStatementNum, statements);
+    if (!badIndentationStatements.isEmpty()) {
+      String message = String.format(statements.get(nestingStatementNum).is(Kind.IF_STATEMENT) ? MESSAGE_IF : MESSAGE_LOOP, 1 + badIndentationStatements.size());
+      PreciseIssue issue = context().newIssue(this, badIndentationStatements.get(0), message);
+      issue.secondary(firstInnerStatement, null);
+      badIndentationStatements.stream().skip(1).forEach(statement -> issue.secondary(statement, null));
+    }
+  }
 
-    int firstIndent = column(firstInnerStatement);
-    List<StatementTree> statementsWhichShouldBeNested = new ArrayList<>();
-
+  private static List<StatementTree> badIndentationStatements(int headerLine, int headerColumn, StatementTree firstInnerStatement, int nestingStatementNum,
+    List<StatementTree> statements) {
+    List<StatementTree> badStatements = new ArrayList<>();
+    int lastStatementLine = line(firstInnerStatement);
+    int lastStatementColumn = column(firstInnerStatement);
     for (int i = nestingStatementNum + 1; i < statements.size(); i++) {
-      StatementTree nextStatement = statements.get(i);
-
-      if (!nextStatement.is(Kind.INLINE_HTML)) {
-
-        if (column(nextStatement) == firstIndent) {
-          statementsWhichShouldBeNested.add(nextStatement);
-
+      StatementTree statement = statements.get(i);
+      if (!statement.is(Kind.INLINE_HTML)) {
+        int line = line(statement);
+        int column = column(statement);
+        if (line == lastStatementLine) {
+          badStatements.add(statement);
+        } else if (((column != headerColumn && line < (lastStatementLine + 4)) ||
```

Comment:
No magic number! Please use a dedicated constant to explain what it is for-

Reply:
In fact I would use for each of these tests a dedicated method with a proper name:
* `(column != headerColumn && line < (lastStatementLine + 4))`
* `(column == headerColumn && line == (lastStatementLine + 1))`